closed/src/java.base/zos/native/libsyslookup/syslookup.c

```diff
@@ -0,0 +1,325 @@
+#endif /* _OPEN_SYS_DIR_EXT */
+#ifndef _OPEN_SYS_FILE_EXT
+#define _OPEN_SYS_FILE_EXT
+#endif /* _OPEN_SYS_FILE_EXT */
+#ifndef _OPEN_SYS_PTY_EXTENSIONS
+#define _OPEN_SYS_PTY_EXTENSIONS
+#endif /* _OPEN_SYS_PTY_EXTENSIONS */
+#ifndef _OPEN_SYS_SOCK_EXT
+#define _OPEN_SYS_SOCK_EXT
+#endif /* _OPEN_SYS_SOCK_EXT */
+#ifndef _OPEN_SYS_SOCK_EXT2
+#define _OPEN_SYS_SOCK_EXT2
+#endif /* _OPEN_SYS_SOCK_EXT2 */
+#ifndef _OPEN_SYS_SOCK_EXT3
+#define _OPEN_SYS_SOCK_EXT3
+#endif /* _OPEN_SYS_SOCK_EXT3 */
+#ifndef _OPEN_SYS_SOCK_IPV6
+#define _OPEN_SYS_SOCK_IPV6
+#endif /* _OPEN_SYS_SOCK_IPV6 */
+#ifndef _OPEN_SYS_TIMED_EXT
+#define _OPEN_SYS_TIMED_EXT 1
+#endif /* _OPEN_SYS_TIMED_EXT */
+#ifndef _OPEN_THREADS
+#define _OPEN_THREADS
+#endif /* _OPEN_THREADS */
+#ifndef _POSIX_SOURCE
+#define _POSIX_SOURCE
+#endif /* _POSIX_SOURCE */
+#ifndef _UNIX03_SOURCE
+#define _UNIX03_SOURCE
+#endif /* _UNIX03_SOURCE */
+#ifndef _XOPEN_SOURCE
+#define _XOPEN_SOURCE 500
+#endif /* _XOPEN_SOURCE */
+#ifndef _XOPEN_SOURCE_EXTENDED
+#define _XOPEN_SOURCE_EXTENDED 1
+#endif /* _XOPEN_SOURCE_EXTENDED */
+
+#include <arpa/inet.h>
+#include <arpa/nameser.h>
+#include <assert.h>
+#include <collate.h>
+#include <complex.h>
+#include <ctest.h>
+#include <ctype.h>
+#include <dirent.h>
+#include <dll.h>
+#include <dynit.h>
+#include <fcntl.h>
+#include <fenv.h>
+#include <float.h>
+#include <fmtmsg.h>
+#include <fnmatch.h>
+#include <fpxcp.h>
+#include <ftw.h>
+#include <glob.h>
+#include <grp.h>
+#include <iconv.h>
+#include <langinfo.h>
+#include <libgen.h>
+#include <locale.h>
+#include <math.h>
+#include <monetary.h>
+#include <ndbm.h>
+#include <netdb.h>
+#include <netinet/in.h>
+#include <net/if.h>
+#include <nlist.h>
+#include <nl_types.h>
+#include <poll.h>
+#include <pthread.h>
+#include <pwd.h>
+#include <regex.h>
+#include <resolv.h>
+#include <rexec.h>
+#include <search.h>
+#include <setjmp.h>
+#include <signal.h>
+#include <spawn.h>
+#include <stdio.h>
+#include <stdio_ext.h>
+#include <stdlib.h>
+#include <string.h>
+#include <strings.h>
+#include <stropts.h>
+#include <sys/acl.h>
+#include <sys/ipc.h>
+#include <sys/layout.h>
+#include <sys/mman.h>
+#include <sys/msg.h>
+#include <sys/ps.h>
+#include <sys/resource.h>
+#include <sys/sem.h>
+#include <sys/shm.h>
+#include <sys/socket.h>
+#include <sys/stat.h>
+#include <sys/statfs.h>
+#include <sys/statvfs.h>
+#include <sys/time.h>
+#include <sys/timeb.h>
+#include <sys/times.h>
+#include <sys/types.h>
+#include <sys/utsname.h>
+#include <sys/wait.h>
+#include <sys/__cpl.h>
+#include <sys/__messag.h>
+#include <sys/__wlm.h>
+#include <syslog.h>
+#include <termios.h>
+#include <time.h>
+#include <ucontext.h>
+#include <uheap.h>
+#include <ulimit.h>
+#include <unistd.h>
+#include <utime.h>
+#include <utmpx.h>
+#include <variant.h>
+#include <wchar.h>
+#include <wctype.h>
+#include <wordexp.h>
+
+#ifdef T_NULL
+#undef T_NULL
+#endif /* T_NULL */
+#ifdef T_UNSPEC
+#undef T_UNSPEC
+#endif /* T_UNSPEC */
```

Comment:
A comment would be welcome explaining why we don't want these symbols defined.

Reply:
Comment is added as suggested.

Reply:
The conflict was caused by the definitions used by both headers as follows:
```
/usr/include/arpa/nameser.h
153  #define T_NULL          10              /* null resource record */
166  #define T_UNSPEC        103             /* Unspecified format (binary data) */

/usr/include/xti.h
311  #define T_UNUSED        -1
312  #define T_NULL           0
```
in which case reordering the includes doesn't help to bypass the case (you have to `undefine` them after one header before using in the other header).